### PR TITLE
Refine code

### PR DESCRIPTION
**Pull request by Goooler**
## :page_facing_up: Context

Some code sections should be refined, could be replaced with properer ones.

## :pencil: Changes

- Remove redundant comma.
- Replace `?: ""` with `orEmpty()`.
- Add params with default value.
- Use `LiveData#Observe()` SAM syntax.
And more...
## :no_entry_sign: Breaking

No.
## :hammer_and_wrench: How to test

No functional changes
## :stopwatch: Next steps

N/A.

### CODE REVIEW

**Comment by MiSikora** [comment]
Please use our PR template and remove sections only if they are irrelevant to changes.

**Comment by Goooler** [comment]
> Please use our PR template and remove sections only if they are irrelevant to changes.

Can you review it now ?

File: library/src/main/java/com/chuckerteam/chucker/internal/data/repository/HttpTransactionDatabaseRepository.kt [hunk 1]
```diff
@@ -4,11 +4,13 @@ import androidx.lifecycle.LiveData
...
 import com.chuckerteam.chucker.internal.data.room.ChuckerDatabase
+import com.chuckerteam.chucker.internal.data.room.HttpTransactionDao
 import com.chuckerteam.chucker.internal.support.distinctUntilChanged
 
-internal class HttpTransactionDatabaseRepository(private val database: ChuckerDatabase) : HttpTransactionRepository {
+internal class HttpTransactionDatabaseRepository(database: ChuckerDatabase) :
+    HttpTransactionRepository {
 
-    private val transactionDao get() = database.transactionDao()
+    private val transactionDao: HttpTransactionDao = database.transactionDao
```
Comment (MiSikora) [inline library/src/main/java/com/chuckerteam/chucker/internal/data/repository/HttpTransactionDatabaseRepository.kt:13]: Why this change?

File: library/src/main/java/com/chuckerteam/chucker/internal/data/room/ChuckerDatabase.kt [hunk 2]
```diff
@@ -9,7 +9,7 @@ import com.chuckerteam.chucker.internal.data.entity.HttpTransaction
 @Database(entities = [HttpTransaction::class], version = 6, exportSchema = false)
 internal abstract class ChuckerDatabase : RoomDatabase() {
 
-    abstract fun transactionDao(): HttpTransactionDao
+    abstract val transactionDao: HttpTransactionDao
```
Comment (MiSikora) [inline library/src/main/java/com/chuckerteam/chucker/internal/data/room/ChuckerDatabase.kt:12]: Same here, why change this?

File: library/src/main/java/com/chuckerteam/chucker/internal/support/ContextExt.kt [hunk 3]
```diff
@@ -6,17 +6,17 @@ import com.google.android.material.dialog.MaterialAlertDialogBuilder
 
 internal fun Context.showDialog(
     dialogData: DialogData,
-    onPositiveClick: (() -> Unit)?,
-    onNegativeClick: (() -> Unit)?
+    onPositiveClick: () -> Unit = {},
+    onNegativeClick: () -> Unit = {}
```
Comment (MiSikora) [inline library/src/main/java/com/chuckerteam/chucker/internal/support/ContextExt.kt:10]: I'm ok with default arguments but why make them not nullable?

File: library/src/main/java/com/chuckerteam/chucker/internal/data/repository/HttpTransactionDatabaseRepository.kt (hunk 1 above)
Comment (Goooler) [reply library/src/main/java/com/chuckerteam/chucker/internal/data/repository/HttpTransactionDatabaseRepository.kt:13]: Removed val on the `database` in consctructor, transactionDao can be initiallized directly when constructing.

File: library/src/main/java/com/chuckerteam/chucker/internal/data/room/ChuckerDatabase.kt (hunk 2 above)
Comment (Goooler) [reply library/src/main/java/com/chuckerteam/chucker/internal/data/room/ChuckerDatabase.kt:12]: Kotlin feature, use`val` replace `fun` in`interface` or `abstract class`.

File: library/src/main/java/com/chuckerteam/chucker/internal/data/repository/HttpTransactionDatabaseRepository.kt (hunk 1 above)
Comment (MiSikora) [reply library/src/main/java/com/chuckerteam/chucker/internal/data/repository/HttpTransactionDatabaseRepository.kt:13]: Sorry, I don't get this response. Which constructor and which initialization do you have on mind? DAOs are lazy initialized and synchronized in Room.

File: library/src/main/java/com/chuckerteam/chucker/internal/data/room/ChuckerDatabase.kt (hunk 2 above)
Comment (MiSikora) [reply library/src/main/java/com/chuckerteam/chucker/internal/data/room/ChuckerDatabase.kt:12]: I don't know why would you like to change it. There is nothing wrong or unidiomatic with using a function.

File: library/src/main/java/com/chuckerteam/chucker/internal/support/ContextExt.kt (hunk 3 above)
Comment (Goooler) [reply library/src/main/java/com/chuckerteam/chucker/internal/support/ContextExt.kt:10]: If the arguments are nullable and with default value, look like `(() -> Unit)? = null`, it's chatty and not neat.

Comment (MiSikora) [reply library/src/main/java/com/chuckerteam/chucker/internal/support/ContextExt.kt:10]: What's chatty and not neat and by which metric? Having nullable arguments saves object allocations.

File: library/src/main/java/com/chuckerteam/chucker/internal/data/repository/HttpTransactionDatabaseRepository.kt (hunk 1 above)
Comment (Goooler) [reply library/src/main/java/com/chuckerteam/chucker/internal/data/repository/HttpTransactionDatabaseRepository.kt:13]: image
Here, `database` is just a injected param only provide `dao` for `transactionDao`

File: library/src/main/java/com/chuckerteam/chucker/internal/data/room/ChuckerDatabase.kt (hunk 2 above)
Comment (Goooler) [reply library/src/main/java/com/chuckerteam/chucker/internal/data/room/ChuckerDatabase.kt:12]: Yeah, behaves in the same way. But I think use a field instead of function here is more graceful.

File: library/src/main/java/com/chuckerteam/chucker/internal/data/repository/HttpTransactionDatabaseRepository.kt (hunk 1 above)
Comment (MiSikora) [reply library/src/main/java/com/chuckerteam/chucker/internal/data/repository/HttpTransactionDatabaseRepository.kt:13]: Ah, thanks. Didn't catch that. Why would it be a problem to keep a reference to the database instead of DAO?

File: library/src/main/java/com/chuckerteam/chucker/internal/support/ContextExt.kt (hunk 3 above)
Comment (Goooler) [reply library/src/main/java/com/chuckerteam/chucker/internal/support/ContextExt.kt:10]: Ok, for needing not to allocate objects, I will change them to nullable.

File: library/src/main/java/com/chuckerteam/chucker/internal/data/room/ChuckerDatabase.kt (hunk 2 above)
Comment (MiSikora) [reply library/src/main/java/com/chuckerteam/chucker/internal/data/room/ChuckerDatabase.kt:12]: This is just opinion based. I don't mind the output code but I don't like to change the code just for cosmetic changes when there are no objective indicators to do so.

File: library/src/main/java/com/chuckerteam/chucker/internal/data/repository/HttpTransactionDatabaseRepository.kt (hunk 1 above)
Comment (Goooler) [reply library/src/main/java/com/chuckerteam/chucker/internal/data/repository/HttpTransactionDatabaseRepository.kt:13]: `database` just used to provide `dao` for `transactionDao`, we can remove the reference after constructor for min visibility.

File: library/src/main/java/com/chuckerteam/chucker/internal/data/room/ChuckerDatabase.kt (hunk 2 above)
Comment (Goooler) [reply library/src/main/java/com/chuckerteam/chucker/internal/data/room/ChuckerDatabase.kt:12]: Another reason, for Java code to access it, we usually use `getTransactionDao()`, but in Kotlin we are suggested to use `transactionDao`. What I modified can be more compatible with Java and Kotlin.

Comment (MiSikora) [reply library/src/main/java/com/chuckerteam/chucker/internal/data/room/ChuckerDatabase.kt:12]: This would be true, if this class was exposed. Don't get me wrong, I'm not requesting for you to change it back to a function but it is not something that refines the code. It is just formatting it according to personal preferences.

File: library/src/main/java/com/chuckerteam/chucker/internal/data/repository/HttpTransactionDatabaseRepository.kt (hunk 1 above)
Comment (MiSikora) [reply library/src/main/java/com/chuckerteam/chucker/internal/data/repository/HttpTransactionDatabaseRepository.kt:13]: Visibility is still the same. `HttpTransactionDatabaseRepository` has access to database anyway. More meaningful would be to pass DAO as a constructor parameter.

**Comment by vbuberen** [comment]
After looking at changes in this PR I see that most things changed are just opinion based, not something really meaningful and really that better/worse depends on someone's preference. 
While shortcut feature was a real enhancement this PR looks like some change just for the sake of change.

@Goooler I would like to ask you to stop doing such PRs, since the value of it is quite questionable (like dependencies update a few days ago or changing callbacks from nullable to not-nullable to only return it back later in the same PR.

File: library/src/main/java/com/chuckerteam/chucker/internal/data/repository/HttpTransactionDatabaseRepository.kt (hunk 1 above)
Comment (Goooler) [reply library/src/main/java/com/chuckerteam/chucker/internal/data/repository/HttpTransactionDatabaseRepository.kt:13]: Yeah, it's only duty is as a constructor parameter, should not let other functions to access.

**Comment by Goooler** [comment]
@vbuberen Thanks for your patience.

**Comment by MiSikora** [comment]
For the reasons mentioned by @vbuberen with which I agree. I'm going to close this PR. These changes feel mostly opinionated without a lot of substance. Don't feel refrained from making PRs, but next time maybe try with issues from the backlog and not only cosmetic changes. Also, if you have ideas it is usually better to start with an issue.

**Comment by Goooler** [comment]
> For the reasons mentioned by @vbuberen with which I agree, I'm going to close this PR. These changes feel mostly opinionated without a lot of substance. Don't feel refrained from making PRs, but next time maybe try with issues from the backlog and not only cosmetic changes. Also, if you have ideas it is usually better to start with an issue so we can talk it through.

Thanks for your patience.
